pandas/tests/reshape/test_crosstab.py

```diff
@@ -743,3 +743,27 @@ def test_margin_normalize_multiple_columns(self):
         )
         expected.index.name = "C"
         tm.assert_frame_equal(result, expected)
+
+
+def test_categoricals():
+    # https://github.com/pandas-dev/pandas/issues/37465
+    g = np.random.RandomState(25982704)
```

Comment:
it might be worth parameterize this to check if a is a cat and b is not (and vice versa), might be slightly tricky.